text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+# Context Propagation: A Layered Approach 
+
+* [Motivation](#Motivation)
+* [OpenTelemetry layered architecture](#OpenTelemetry-layered-architecture)
+  * [Cross-Cutting Concerns](#Cross-Cutting-Concerns)
+    * [Observability API](#Observability-API)
+    * [Correlations API](#Correlations-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+  * [Global initialization](#Global-initialization)
+  * [Extracting and injecting from HTTP headers](#Extracting-and-injecting-from-HTTP-headers)
+  * [Simplify the API with automated context propagation](#Simplify-the-API-with-automated-context-propagation)
+  * [Implementing a propagator](#Implementing-a-propagator)
+  * [Implementing a concern](#Implementing-a-concern)
+  * [The scope of current context](#The-scope-of-current-context)
+  * [Referencing multiple contexts](#Referencing-multiple-contexts)
+  * [Falling back to explicit contexts](#Falling-back-to-explicit-contexts)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+# Motivation
+
+This RFC addresses the following topics:
+
+**Separation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Correlations) and observability data (SpanContext, etc) 
```

Comment:
separately from what? `DistributedContext` class defined in spec is already separate from the `Tracer`.

Reply:
It is not actually that different from what we have now. This change looks bigger than it actually is, imho. :)

I will think about how to further clean up the justifications.